### ISSUE TITLE
[Bug]: Unable to update background execution mode: `conflict between new type (mixed) and old type (string)`

### ISSUE THREAD

**Issue by theyoungstone** [nextcloud/server#45083]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

When I want to change the background task execution mode, I have the message "Impossible de mettre à jour le mode d'exécution des tâches d'arrière-plan" (Unable to update background task execution mode)

### Steps to reproduce

1. Go to "Administration parameters"
2. Go to "Basic settings"
3. Try to change background task execution mode from "Cron" to another

### Expected behavior

Change accepted

### Installation method

Community Manual Installation with Archive

### Nextcloud Server version

29

### Operating system

Debian

### PHP engine version

8.3

### Web server

Apache

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Update from 28.0.5 to 29.0.0

### Are you using the Nextcloud Server Encryption module?

No

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud.lejeune.pl"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "29.0.0.19",
        "overwrite.cli.url": "https:\/\/nextcloud.lejeune.pl",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Memcached",
        "memcache.locking": "\\OC\\Memcache\\Memcached",
        "memcached_servers": [
            [
                "localhost",
                11211
            ]
        ],
        "maintenance": false,
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpsecure": "ssl",
        "has_rebuilt_cache": true,
        "twofactor_enforced": "true",
        "twofactor_enforced_groups": [
            "DoubleAuth"
        ],
        "twofactor_enforced_excluded_groups": [
            "SimpleAuth"
        ],
        "theme": "",
        "loglevel": 2,
        "updater.release.channel": "stable",
        "app_install_overwrite": [
            "calendar",
            "ransomware_detection",
            "gpxedit",
            "gpxpod",
            "phonetrack",
            "quota_warning",
            "ransomware_protection"
        ],
        "default_phone_region": "FR",
        "maintenance_window_start": "1",
        "trashbin_retention_obligation": "auto",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    },
    "apps": {
        "activity": {
            "enable_email": "no",
            "enabled": "yes",
            "installed_version": "2.21.1",
            "notify_email_calendar": "0",
            "notify_email_calendar_event": "0",
            "notify_email_calendar_todo": "0",
            "notify_email_comments": "0",
            "notify_email_deck": "0",
            "notify_email_deck_card_description": "0",
            "notify_email_deck_comment": "0",
            "notify_email_favorite": "0",
            "notify_email_file_changed": "0",
            "notify_email_file_created": "0",
            "notify_email_file_deleted": "0",
            "notify_email_file_favorite_changed": "0",
            "notify_email_file_restored": "0",
            "notify_email_phonetrack": "0",
            "notify_email_phonetrack_geofence_event": "0",
            "notify_email_phonetrack_proximity_event": "0",
            "notify_email_public_links": "1",
            "notify_email_remote_share": "0",
            "notify_email_shared": "0",
            "notify_email_systemtags": "0",
            "notify_notification_calendar": "0",
            "notify_notification_calendar_event": "0",
            "notify_notification_calendar_todo": "0",
            "notify_notification_comments": "0",
            "notify_notification_deck": "0",
            "notify_notification_deck_card_description": "0",
            "notify_notification_deck_comment": "0",
            "notify_notification_favorite": "0",
            "notify_notification_file_changed": "0",
            "notify_notification_file_favorite_changed": "1",
            "notify_notification_group_settings": "1",
            "notify_notification_personal_settings": "0",
            "notify_notification_phonetrack": "0",
            "notify_notification_phonetrack_geofence_event": "0",
            "notify_notification_phonetrack_proximity_event": "0",
            "notify_notification_public_links": "0",
            "notify_notification_remote_share": "0",
            "notify_notification_security": "1",
            "notify_notification_shared": "1",
            "notify_notification_systemtags": "0",
            "notify_notification_twofactor_totp": "1",
            "notify_setting_batchtime": "3600",
            "notify_setting_self": "0",
            "notify_setting_selfemail": "0",
            "notify_stream_calendar": "1",
            "notify_stream_calendar_event": "1",
            "notify_stream_calendar_todo": "1",
            "notify_stream_comments": "1",
            "notify_stream_deck": "1",
            "notify_stream_deck_card_description": "1",
            "notify_stream_favorite": "1",
            "notify_stream_file_changed": "1",
            "notify_stream_file_created": "1",
            "notify_stream_file_deleted": "1",
            "notify_stream_file_favorite": "0",
            "notify_stream_file_restored": "1",
            "notify_stream_phonetrack": "1",
            "notify_stream_phonetrack_geofence_event": "1",
            "notify_stream_phonetrack_proximity_event": "1",
            "notify_stream_public_links": "1",
            "notify_stream_remote_share": "1",
            "notify_stream_shared": "1",
            "notify_stream_systemtags": "1",
            "types": "filesystem"
        },
        "backgroundjob": {
            "lastjob": "19086"
        },
        "bruteForce": {
            "whitelist_2": "82.64.57.215\/32"
        },
        "bruteforcesettings": {
            "enabled": "yes",
            "installed_version": "2.9.0",
            "types": ""
        },
        "calendar": {
            "enabled": "yes",
            "installed_version": "4.7.1",
            "types": ""
        },
        "carnet": {
            "enabled": "no",
            "installed_version": "0.20.0",
            "types": "filesystem"
        },
        "circles": {
            "enabled": "yes",
            "installed_version": "29.0.0-dev",
            "loopback_tmp_scheme": "https",
            "maintenance_run": "0",
            "maintenance_update": "{\"maximum\":3,\"3\":1714238102,\"2\":1714239003,\"1\":1714239003,\"4\":1652320802,\"5\":1651892403}",
            "migration_22": "1",
            "migration_22_1": "1",
            "migration_run": "0",
            "types": "filesystem,dav"
        },
        "cloud_federation_api": {
            "enabled": "yes",
            "installed_version": "1.12.0",
            "types": "filesystem"
        },
        "comments": {
            "enabled": "yes",
            "installed_version": "1.19.0",
            "types": "logging"
        },
        "contacts": {
            "enabled": "yes",
            "installed_version": "6.0.0",
            "types": "dav"
        },
        "contactsinteraction": {
            "enabled": "yes",
            "installed_version": "1.10.0",
            "types": "dav"
        },
        "core": {
            "backgroundjobs_mode": "cron",
            "enterpriseLogoChecked": "yes",
            "files_metadata": "{\"photos-original_date_time\":{\"value\":null,\"type\":\"int\",\"indexed\":true,\"editPermission\":0},\"photos-exif\":{\"value\":null,\"type\":\"array\",\"indexed\":false,\"editPermission\":0},\"photos-ifd0\":{\"value\":null,\"type\":\"array\",\"indexed\":false,\"editPermission\":0},\"photos-size\":{\"value\":null,\"type\":\"array\",\"indexed\":false,\"editPermission\":0},\"photos-gps\":{\"value\":null,\"type\":\"array\",\"indexed\":false,\"editPermission\":0},\"photos-place\":{\"value\":null,\"type\":\"string\",\"indexed\":true,\"editPermission\":0},\"files-live-photo\":{\"value\":null,\"type\":\"string\",\"indexed\":false,\"editPermission\":1}}",
            "files_metadata_installed": "1",
            "installed.bundles": "[\"CoreBundle\"]",
            "installedat": "1584987245.804",
            "lastcron": "1714283220",
            "lastupdateResult": "[]",
            "lastupdatedat": "1714283141",
            "moveavatarsdone": "yes",
            "previewsCleanedUp": "1",
            "public_files": "files_sharing\/public.php",
            "public_webdav": "dav\/appinfo\/v1\/publicwebdav.php",
            "theming.variables": "3380c74585402bc749599a676554ab7b",
            "updater.secret.created": "1714238927",
            "vendor": "nextcloud",
            "oc.integritycheck.checker": "[]"
        },
        "cospend": {
            "enabled": "no",
            "installed_version": "1.3.7",
            "types": ""
        },
        "dashboard": {
            "enabled": "no",
            "installed_version": "7.1.0",
            "types": ""
        },
        "dav": {
            "buildCalendarReminderIndex": "yes",
            "buildCalendarSearchIndex": "yes",
            "builtSocialSearchIndex": "yes",
            "chunks_migrated": "1",
            "enabled": "yes",
            "installed_version": "1.30.1",
            "regeneratedBirthdayCalendarsForYearFix": "yes",
            "types": "filesystem"
        },
        "deck": {
            "enabled": "no",
            "installed_version": "1.3.2",
            "types": "dav"
        },
        "duplicatefinder": {
            "enabled": "no",
            "installed_version": "0.0.15",
            "types": ""
        },
        "federatedfilesharing": {
            "enabled": "yes",
            "installed_version": "1.19.0",
            "types": ""
        },
        "federation": {
            "enabled": "yes",
            "installed_version": "1.19.0",
            "types": "authentication"
        },
        "files": {
            "cronjob_scan_files": "500",
            "default_quota": "1 GB",
            "enabled": "yes",
            "installed_version": "2.1.0",
            "types": "filesystem"
        },
        "files_antivirus": {
            "av_path": "\/usr\/bin\/clamscan",
            "enabled": "no",
            "installed_version": "3.0.0",
            "types": "filesystem,dav"
        },
        "files_downloadlimit": {
            "enabled": "yes",
            "installed_version": "2.0.0",
            "types": ""
        },
        "files_pdfviewer": {
            "enabled": "yes",
            "installed_version": "2.10.0",
            "types": ""
        },
        "files_reminders": {
            "enabled": "yes",
            "installed_version": "1.2.0",
            "types": ""
        },
        "files_retention": {
            "enabled": "no",
            "installed_version": "1.10.0",
            "types": "filesystem"
        },
        "files_rightclick": {
            "enabled": "no",
            "installed_version": "1.6.0",
            "types": ""
        },
        "files_sharing": {
            "enabled": "yes",
            "installed_version": "1.21.0",
            "types": "filesystem"
        },
        "files_trashbin": {
            "enabled": "yes",
            "installed_version": "1.19.0",
            "types": "filesystem,dav"
        },
        "files_versions": {
            "enabled": "yes",
            "installed_version": "1.22.0",
            "types": "filesystem,dav"
        },
        "firstrunwizard": {
            "enabled": "yes",
            "installed_version": "2.18.0",
            "types": "logging"
        },
        "gpxedit": {
            "enabled": "no",
            "installed_version": "0.0.14",
            "types": ""
        },
        "gpxmotion": {
            "enabled": "no",
            "installed_version": "0.1.0",
            "types": ""
        },
        "gpxpod": {
            "enabled": "yes",
            "installed_version": "5.0.18",
            "reset304": "1",
            "resetPics404": "1",
            "types": ""
        },
        "groupfolders": {
            "enabled": "no",
            "installed_version": "16.0.6",
            "types": "filesystem,dav"
        },
        "logreader": {
            "enabled": "yes",
            "installed_version": "2.14.0",
            "levels": "00111",
            "shownLevels": "[2,3,4]",
            "types": "logging"
        },
        "lookup_server_connector": {
            "enabled": "yes",
            "installed_version": "1.17.0",
            "types": "authentication"
        },
        "mail": {
            "enabled": "yes",
            "installed_version": "3.6.0",
            "types": ""
        },
        "maps": {
            "enabled": "no",
            "installed_version": "0.1.6",
            "lastAddressLookup": "1585295213",
            "types": "filesystem"
        },
        "nextcloud_announcements": {
            "enabled": "yes",
            "installed_version": "1.18.0",
            "pub_date": "Thu, 24 Oct 2019 00:00:00 +0200",
            "types": "logging"
        },
        "notes": {
            "enabled": "yes",
            "installed_version": "4.10.0",
            "types": ""
        },
        "notifications": {
            "enabled": "yes",
            "installed_version": "2.17.0",
            "types": "logging"
        },
        "oauth2": {
            "enabled": "yes",
            "installed_version": "1.17.0",
            "types": "authentication"
        },
        "password_policy": {
            "enabled": "yes",
            "enforceHaveIBeenPwned": "1",
            "enforceNumericCharacters": "1",
            "enforceSpecialCharacters": "1",
            "enforceUpperLowerCase": "1",
            "installed_version": "1.19.0",
            "types": "authentication"
        },
        "phonetrack": {
            "enabled": "yes",
            "installed_version": "0.7.7",
            "pointQuota": "20000",
            "types": ""
        },
        "photos": {
            "enabled": "yes",
            "installed_version": "2.5.0",
            "lastPlaceMappedUser": "pierre",
            "lastPlaceMappingDone": "true",
            "types": "dav,authentication"
        },
        "privacy": {
            "enabled": "yes",
            "installed_version": "1.13.0",
            "readableLocation": "fr",
            "types": ""
        },
        "provisioning_api": {
            "enabled": "yes",
            "installed_version": "1.19.0",
            "types": "prevent_group_restriction"
        },
        "quota_warning": {
            "alert_email": "yes",
            "enabled": "yes",
            "info_email": "yes",
            "installed_version": "1.19.0",
            "types": "filesystem",
            "warning_email": "yes"
        },
        "ransomware_detection": {
            "debug": "0",
            "enabled": "no",
            "installed_version": "0.10.0",
            "types": ""
        },
        "ransomware_protection": {
            "enabled": "no",
            "installed_version": "1.14.0",
            "types": "logging"
        },
        "recommendations": {
            "enabled": "yes",
            "installed_version": "2.1.0",
            "types": ""
        },
        "related_resources": {
            "enabled": "yes",
            "installed_version": "1.4.0",
            "types": ""
        },
        "serverinfo": {
            "cached_count_filecache": "59169",
            "cached_count_storages": "6",
            "enabled": "yes",
            "installed_version": "1.19.0",
            "types": ""
        },
        "settings": {
            "enabled": "yes",
            "installed_version": "1.12.0",
            "types": ""
        },
        "sharebymail": {
            "enabled": "yes",
            "installed_version": "1.19.0",
            "types": "filesystem"
        },
        "side_menu": {
            "add-logo-link": "0",
            "always-displayed": "0",
            "background-color": "#333333",
            "background-color-opacity": "100",
            "background-color-to": "#333333",
            "big-menu": "0",
            "big-menu-hidden-apps": "[]",
            "cache": "11",
            "cache-categories": <Comment to long for bug formular>,
            "categories-order": "[\"other\",\"office\",\"external_links\",\"files\",\"workflow\",\"integration\",\"games\",\"multimedia\",\"organization\",\"tools\",\"customization\",\"search\",\"social\",\"monitoring\",\"security\",\"dashboard\"]",
            "categories-order-type": "default",
            "current-app-background-color": "#444444",
            "dark-mode-background-color": "#333333",
            "dark-mode-background-color-opacity": "100",
            "dark-mode-background-color-to": "#333333",
            "dark-mode-current-app-background-color": "#444444",
            "dark-mode-icon-invert-filter": "0",
            "dark-mode-icon-opacity": "100",
            "dark-mode-loader-color": "#cccccc",
            "dark-mode-opener": "side-menu-opener",
            "dark-mode-text-color": "#ffffff",
            "default-enabled": "1",
            "display-logo": "1",
            "enabled": "no",
            "external-sites-in-top-menu": "0",
            "force": "0",
            "force-light-icon": "0",
            "hide-when-no-apps": "1",
            "icon-invert-filter": "0",
            "icon-opacity": "0",
            "installed_version": "2.0.1",
            "loader-color": "#0e75ac",
            "loader-enabled": "1",
            "opener": "side-menu-opener",
            "opener-hover": "0",
            "opener-only": "0",
            "opener-position": "before",
            "show-settings": "0",
            "side-with-categories": "0",
            "size-icon": "small",
            "size-text": "normal",
            "target-blank-apps": "[]",
            "text-color": "#ffffff",
            "top-menu-apps": "[]",
            "types": "",
            "undefined": "",
            "use-avatar": "0"
        },
        "support": {
            "SwitchUpdaterServerHasRun": "yes",
            "enabled": "yes",
            "installed_version": "1.12.0",
            "types": "session"
        },
        "survey_client": {
            "enabled": "yes",
            "installed_version": "1.17.0",
            "last_report": "{\"id\":\"ocfdzy5tv71g\",\"items\":[[\"server\",\"version\",\"28.0.4.1\"],[\"server\",\"code\",\"other\"],[\"server\",\"enable_avatars\",\"yes\"],[\"server\",\"enable_previews\",\"yes\"],[\"server\",\"memcache.local\",\"\\\\OC\\\\Memcache\\\\APCu\"],[\"server\",\"memcache.distributed\",\"\\\\OC\\\\Memcache\\\\Memcached\"],[\"server\",\"asset-pipeline.enabled\",\"no\"],[\"server\",\"filelocking.enabled\",\"yes\"],[\"server\",\"memcache.locking\",\"\\\\OC\\\\Memcache\\\\Memcached\"],[\"server\",\"debug\",\"no\"],[\"server\",\"cron\",\"cron\"],[\"php\",\"version\",\"8.3.6\"],[\"php\",\"memory_limit\",-1],[\"php\",\"max_execution_time\",0],[\"php\",\"upload_max_filesize\",2097152],[\"database\",\"type\",\"mysql\"],[\"database\",\"version\",\"10.3.39\"],[\"database\",\"size\",99917824],[\"apps\",\"activity\",\"2.20.0\"],[\"apps\",\"bruteforcesettings\",\"2.8.0\"],[\"apps\",\"calendar\",\"4.6.7\"],[\"apps\",\"carnet\",\"disabled\"],[\"apps\",\"circles\",\"28.0.0\"],[\"apps\",\"cloud_federation_api\",\"1.11.0\"],[\"apps\",\"comments\",\"1.18.0\"],[\"apps\",\"contacts\",\"5.5.3\"],[\"apps\",\"contactsinteraction\",\"1.9.0\"],[\"apps\",\"cospend\",\"disabled\"],[\"apps\",\"dashboard\",\"disabled\"],[\"apps\",\"dav\",\"1.29.1\"],[\"apps\",\"deck\",\"disabled\"],[\"apps\",\"duplicatefinder\",\"disabled\"],[\"apps\",\"federatedfilesharing\",\"1.18.0\"],[\"apps\",\"federation\",\"1.18.0\"],[\"apps\",\"files\",\"2.0.0\"],[\"apps\",\"files_antivirus\",\"disabled\"],[\"apps\",\"files_pdfviewer\",\"2.9.0\"],[\"apps\",\"files_reminders\",\"1.1.0\"],[\"apps\",\"files_retention\",\"disabled\"],[\"apps\",\"files_rightclick\",\"disabled\"],[\"apps\",\"files_sharing\",\"1.20.0\"],[\"apps\",\"files_trashbin\",\"1.18.0\"],[\"apps\",\"files_versions\",\"1.21.0\"],[\"apps\",\"firstrunwizard\",\"2.17.0\"],[\"apps\",\"gpxedit\",\"disabled\"],[\"apps\",\"gpxmotion\",\"disabled\"],[\"apps\",\"gpxpod\",\"5.0.18\"],[\"apps\",\"groupfolders\",\"16.0.6\"],[\"apps\",\"logreader\",\"2.13.0\"],[\"apps\",\"lookup_server_connector\",\"1.16.0\"],[\"apps\",\"mail\",\"3.5.7\"],[\"apps\",\"maps\",\"disabled\"],[\"apps\",\"nextcloud_announcements\",\"1.17.0\"],[\"apps\",\"notes\",\"4.9.4\"],[\"apps\",\"notifications\",\"2.16.0\"],[\"apps\",\"oauth2\",\"1.16.3\"],[\"apps\",\"password_policy\",\"1.18.0\"],[\"apps\",\"phonetrack\",\"0.7.7\"],[\"apps\",\"photos\",\"2.4.0\"],[\"apps\",\"privacy\",\"1.12.0\"],[\"apps\",\"provisioning_api\",\"1.18.0\"],[\"apps\",\"quota_warning\",\"1.19.0\"],[\"apps\",\"ransomware_detection\",\"disabled\"],[\"apps\",\"ransomware_protection\",\"disabled\"],[\"apps\",\"recommendations\",\"2.0.0\"],[\"apps\",\"related_resources\",\"1.3.0\"],[\"apps\",\"serverinfo\",\"1.18.0\"],[\"apps\",\"settings\",\"1.10.1\"],[\"apps\",\"sharebymail\",\"1.18.0\"],[\"apps\",\"side_menu\",\"disabled\"],[\"apps\",\"support\",\"1.11.1\"],[\"apps\",\"survey_client\",\"1.16.0\"],[\"apps\",\"systemtags\",\"1.18.0\"],[\"apps\",\"tasks\",\"disabled\"],[\"apps\",\"text\",\"3.9.1\"],[\"apps\",\"theming\",\"2.3.0\"],[\"apps\",\"twofactor_backupcodes\",\"1.17.0\"],[\"apps\",\"twofactor_totp\",\"10.0.0-beta.2\"],[\"apps\",\"updatenotification\",\"1.18.0\"],[\"apps\",\"user_status\",\"1.8.1\"],[\"apps\",\"viewer\",\"2.2.0\"],[\"apps\",\"weather_status\",\"1.8.0\"],[\"apps\",\"workflowengine\",\"2.10.0\"],[\"stats\",\"num_files\",58271],[\"stats\",\"num_users\",4],[\"stats\",\"num_storages\",6],[\"stats\",\"num_storages_local\",2],[\"stats\",\"num_storages_home\",4],[\"stats\",\"num_storages_other\",0],[\"stats\",\"num_comments\",0],[\"stats\",\"num_comment_markers\",0],[\"stats\",\"num_systemtags\",0],[\"stats\",\"num_systemtags_mappings\",0],[\"files_sharing\",\"num_shares\",5],[\"files_sharing\",\"num_shares_user\",3],[\"files_sharing\",\"num_shares_groups\",0],[\"files_sharing\",\"num_shares_link\",2],[\"files_sharing\",\"num_shares_link_no_password\",2],[\"files_sharing\",\"num_fed_shares_sent\",0],[\"files_sharing\",\"num_fed_shares_received\",0],[\"files_sharing\",\"permissions_3_17\",2],[\"files_sharing\",\"permissions_0_31\",3],[\"encryption\",\"enabled\",\"no\"],[\"encryption\",\"default_module\",\"no\"]]}",
            "last_sent": "1713354903",
            "server": "yes",
            "types": ""
        },
        "systemtags": {
            "enabled": "yes",
            "installed_version": "1.19.0",
            "types": "logging"
        },
        "tasks": {
            "enabled": "no",
            "installed_version": "0.13.6",
            "types": ""
        },
        "text": {
            "enabled": "yes",
            "installed_version": "3.10.0",
            "types": "dav"
        },
        "theming": {
            "cachebuster": "1",
            "enabled": "yes",
            "installed_version": "2.4.0",
            "types": "logging"
        },
        "twofactor_backupcodes": {
            "enabled": "yes",
            "installed_version": "1.18.0",
            "types": ""
        },
        "twofactor_totp": {
            "enabled": "yes",
            "installed_version": "11.0.0-dev",
            "types": ""
        },
        "updatenotification": {
            "bruteforcesettings": "2.0.1",
            "calendar": "4.6.4",
            "contacts": "5.5.2",
            "core": "28.0.3.2",
            "deck": "1.2.6",
            "enabled": "yes",
            "gpxedit": "0.0.14",
            "gpxpod": "5.0.15",
            "groupfolders": "16.0.4",
            "installed_version": "1.19.1",
            "mail": "3.5.7",
            "notes": "4.9.2",
            "notify_groups": "[\"admin\"]",
            "phonetrack": "0.7.7",
            "quota_warning": "1.18.0",
            "ransomware_detection": "0.10.0",
            "ransomware_protection": "1.13.0",
            "related_resources": "1.0.4",
            "side_menu": "2.0.1",
            "twofactor_totp": "6.4.1",
            "types": "",
            "update_check_errors": "0"
        },
        "user_status": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": ""
        },
        "viewer": {
            "enabled": "yes",
            "installed_version": "2.3.0",
            "types": ""
        },
        "weather_status": {
            "enabled": "yes",
            "installed_version": "1.9.0",
            "types": ""
        },
        "workflowengine": {
            "enabled": "yes",
            "installed_version": "2.11.0",
            "types": "filesystem"
        }
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.21.1
  - bruteforcesettings: 2.9.0
  - calendar: 4.7.1
  - circles: 29.0.0-dev
  - cloud_federation_api: 1.12.0
  - comments: 1.19.0
  - contacts: 6.0.0
  - contactsinteraction: 1.10.0
  - dav: 1.30.1
  - federatedfilesharing: 1.19.0
  - federation: 1.19.0
  - files: 2.1.0
  - files_downloadlimit: 2.0.0
  - files_pdfviewer: 2.10.0
  - files_reminders: 1.2.0
  - files_sharing: 1.21.0
  - files_trashbin: 1.19.0
  - files_versions: 1.22.0
  - firstrunwizard: 2.18.0
  - gpxpod: 5.0.18
  - logreader: 2.14.0
  - lookup_server_connector: 1.17.0
  - mail: 3.6.0
  - nextcloud_announcements: 1.18.0
  - notes: 4.10.0
  - notifications: 2.17.0
  - oauth2: 1.17.0
  - password_policy: 1.19.0
  - phonetrack: 0.7.7
  - photos: 2.5.0
  - privacy: 1.13.0
  - provisioning_api: 1.19.0
  - quota_warning: 1.19.0
  - recommendations: 2.1.0
  - related_resources: 1.4.0
  - serverinfo: 1.19.0
  - settings: 1.12.0
  - sharebymail: 1.19.0
  - support: 1.12.0
  - survey_client: 1.17.0
  - systemtags: 1.19.0
  - text: 3.10.0
  - theming: 2.4.0
  - twofactor_backupcodes: 1.18.0
  - twofactor_totp: 11.0.0-dev
  - updatenotification: 1.19.1
  - user_status: 1.9.0
  - viewer: 2.3.0
  - weather_status: 1.9.0
  - workflowengine: 2.11.0
Disabled:
  - admin_audit: 1.19.0
  - dashboard: 7.9.0 (installed 7.1.0)
  - encryption: 2.17.0
  - files_external: 1.21.0
  - suspicious_login: 7.0.0
  - user_ldap: 1.20.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
{"reqId":"JTeLBOKxg5fEafinsMxV","level":3,"time":"2024-04-28T05:55:49+00:00","remoteAddr":"92.184.99.151","user":"pierre","app":"no app in context","method":"POST","url":"/ocs/v2.php/apps/provisioning_api/api/v1/config/apps/core/backgroundjobs_mode","message":"conflict between new type (mixed) and old type (string)","userAgent":"Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:125.0) Gecko/20100101 Firefox/125.0","version":"29.0.0.19","exception":{"Exception":"OCP\\Exceptions\\AppConfigTypeConflictException","Message":"conflict between new type (mixed) and old type (string)","Code":0,"Trace":[{"file":"/var/www/html/nextcloud/lib/private/AppConfig.php","line":556,"function":"setTypedValue","class":"OC\\AppConfig","type":"->"},{"file":"/var/www/html/nextcloud/apps/provisioning_api/lib/Controller/AppConfigController.php","line":147,"function":"setValueMixed","class":"OC\\AppConfig","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":232,"function":"setValue","class":"OCA\\Provisioning_API\\Controller\\AppConfigController","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php","line":138,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/AppFramework/App.php","line":184,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->"},{"file":"/var/www/html/nextcloud/lib/private/Route/Router.php","line":338,"function":"main","class":"OC\\AppFramework\\App","type":"::"},{"file":"/var/www/html/nextcloud/ocs/v1.php","line":66,"function":"match","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/html/nextcloud/ocs/v2.php","line":23,"args":["/var/www/html/nextcloud/ocs/v1.php"],"function":"require_once"}],"File":"/var/www/html/nextcloud/lib/private/AppConfig.php","Line":821,"message":"conflict between new type (mixed) and old type (string)","exception":[],"CustomMessage":"conflict between new type (mixed) and old type (string)"},"id":"662de84237d0c"}
```

**Comment by SDuesterhaupt** [comment]
If I switchover on my test system (Nextcloud 29, updated over years including beta and RC versions) using the CLI

```
sudo -u apache php /var/www/public_html/nextcloud.domain.tld/occ background:ajax
```

then I the error is shown:

```
In AppConfig.php line 821:

  conflict between new type (mixed) and old type (string)
```

This error does not occur in a productive system (version 29, updated over years excluding beta and RC versions, no fresh install). I assume that a database entry may be out of date.

**Comment by wagnbeu0** [comment]
Can confirm the same issue here on my NC29.0.1 instance.

**Comment by joshtrichards** [comment]
Thanks for reporting. 

Looks possibly related to changes made in PR #41755 that typed the values. 

Not a PR or area of code I'm at all familiar with, but some clues in case someone feels like taking a closer look.

**Comment by Allesanddro** [comment]
Same issue here

**Comment by tobykurien** [comment]
Same here too. Additional info: if I run `./occ background:webcron` to change it, I get:

```
In AppConfig.php line 821:
                                                           
  conflict between new type (mixed) and old type (string)  
                                                           

background:webcron
```

**Comment by HammyHavoc** [comment]
Seeing this too.

**Comment by jarrodjay** [comment]
Same here.

**Comment by HammyHavoc** [comment]
> Same here.

I updated to the latest version of Nextcloud yesterday and it seemed to solve my problem with cron not working and also not being able to switch cron types.

**Comment by theyoungstone** [comment]
> > Same here.
> 
> I updated to the latest version of Nextcloud yesterday and it seemed to solve my problem with cron not working and also not being able to switch cron types.

Nextcloud up to date (stable version : Nextcloud Hub 8 (29.0.2)) and not solved.

**Comment by johnczer** [comment]
Ditto on above

**Comment by CNeutron** [comment]
Same issue here:
`php occ background:ajax `
results in                                                           
```
In AppConfig.php line 821:
                                                           
  conflict between new type (mixed) and old type (string)  
                                                           

background:ajax
```

just updated from 28.0.5 --> 29.0.2

**Comment by johnczer** [comment]
And still happening on 29.0.3:(

**Comment by joshtrichards** [comment]
Rather than posting "me too" comments, upvote :+1: the initial report above, unless you have some new information to add. 

It just spams everybody and wastes developer time needed to fix bugs. Thanks!

**Comment by joshtrichards** [comment]
As a workaround, for the time being to toggle the background mode, you should be able to change the value directly:

```
occ config:app:set --value cron --type string core backgroundjobs_mode 
```

Where `cron` is one of `{ cron, webcron, ajax }`.

It's also possible to force it to be mixed, but I'm hesitant to suggest changing that since I'm not sure what's actually going on here and I still have yet to wrap my head around #41755, the migration path, what the expected behavior was here (other than obviously not breaking the ability to set the background mode).

**Comment by kesselb** [comment]
> The provisioning_api also needs an update to use the types.

The final version should support all types, but the patch below made it work again. 

```
Index: apps/provisioning_api/lib/Controller/AppConfigController.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/apps/provisioning_api/lib/Controller/AppConfigController.php b/apps/provisioning_api/lib/Controller/AppConfigController.php
--- a/apps/provisioning_api/lib/Controller/AppConfigController.php	(revision f6e95d3aec5eb3482891b7ebcd64d6676947d448)
+++ b/apps/provisioning_api/lib/Controller/AppConfigController.php	(date 1719519354804)
@@ -107,7 +107,7 @@
 	 * 200: Value updated successfully
 	 * 403: App or key is not allowed
 	 */
-	public function setValue(string $app, string $key, string $value): DataResponse {
+	public function setValue(string $app, string $key, string $value, int $type = IAppConfig::VALUE_MIXED): DataResponse {
 		$user = $this->userSession->getUser();
 		if ($user === null) {
 			throw new \Exception("User is not logged in."); // Should not happen, since method is guarded by middleware
@@ -124,8 +124,13 @@
 			return new DataResponse(['data' => ['message' => $e->getMessage()]], Http::STATUS_FORBIDDEN);
 		}
 
-		/** @psalm-suppress InternalMethod */
-		$this->appConfig->setValueMixed($app, $key, $value);
+		if ($type === IAppConfig::VALUE_STRING) {
+			$this->appConfig->setValueString($app, $key, $value);
+		} else {
+			/** @psalm-suppress InternalMethod */
+			$this->appConfig->setValueMixed($app, $key, $value);
+		}
+
 		return new DataResponse();
 	}
 
Index: apps/settings/src/components/BasicSettings/BackgroundJob.vue
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/apps/settings/src/components/BasicSettings/BackgroundJob.vue b/apps/settings/src/components/BasicSettings/BackgroundJob.vue
--- a/apps/settings/src/components/BasicSettings/BackgroundJob.vue	(revision f6e95d3aec5eb3482891b7ebcd64d6676947d448)
+++ b/apps/settings/src/components/BasicSettings/BackgroundJob.vue	(date 1719519482717)
@@ -125,7 +125,7 @@
 	},
 	methods: {
 		async onBackgroundJobModeChanged(backgroundJobsMode) {
-			const url = generateOcsUrl('/apps/provisioning_api/api/v1/config/apps/{appId}/{key}', {
+			const url = generateOcsUrl('/apps/provisioning_api/api/v1/config/apps/{appId}/{key}?type=4', {
 				appId: 'core',
 				key: 'backgroundjobs_mode',
 			})

```

**Comment by aronmgv** [comment]
Hello,

I am using crazy-max's nextcloud [image](https://github.com/crazy-max/docker-nextcloud) and was upgrading from `28.0.1` to `29.0.3` and am facing this database [problem ](https://github.com/crazy-max/docker-nextcloud/issues/170) after performing the upgrade. Is it related please?

```
nextcloud.postgres  |
nextcloud.postgres  | PostgreSQL Database directory appears to contain a database; Skipping initialization
nextcloud.postgres  |
nextcloud.postgres  | 2024-07-27 20:31:25.050 UTC [1] LOG:  starting PostgreSQL 12.19 (Debian 12.19-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
nextcloud.postgres  | 2024-07-27 20:31:25.060 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
nextcloud.postgres  | 2024-07-27 20:31:25.060 UTC [1] LOG:  listening on IPv6 address "::", port 5432
nextcloud.postgres  | 2024-07-27 20:31:25.063 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
nextcloud.postgres  | 2024-07-27 20:31:25.070 UTC [26] LOG:  database system was shut down at 2024-07-27 11:14:09 UTC
nextcloud.postgres  | 2024-07-27 20:31:25.079 UTC [1] LOG:  database system is ready to accept connections
nextcloud.postgres  | 2024-07-27 20:31:25.983 UTC [34] ERROR:  column "type" does not exist at character 45
nextcloud.postgres  | 2024-07-27 20:31:25.983 UTC [34] STATEMENT:  SELECT "appid", "configkey", "configvalue", "type" FROM "appconfig" WHERE "lazy" = $1
nextcloud.postgres  | 2024-07-27 20:31:26.323 UTC [38] ERROR:  column "type" does not exist at character 45
nextcloud.postgres  | 2024-07-27 20:31:26.323 UTC [38] STATEMENT:  SELECT "appid", "configkey", "configvalue", "type" FROM "appconfig" WHERE "lazy" = $1
nextcloud.postgres  | 2024-07-27 20:31:26.332 UTC [39] ERROR:  column "type" does not exist at character 45
nextcloud.postgres  | 2024-07-27 20:31:26.332 UTC [39] STATEMENT:  SELECT "appid", "configkey", "configvalue", "type" FROM "appconfig" WHERE "lazy" = $1
```

If so how to use your workaround from above?

I tried and it did not work:

```
root@caradhras:[/docker/nextcloud]: docker-compose exec --user nextcloud nextcloud php occ config:app:set --value cron --type string core backgroundjobs_mode
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
You are about to set config value core/backgroundjobs_mode as STRING

This might break thing, affect performance on your instance or its security!
Confirm this action by typing 'yes': yes
done


In DbalException.php line 71:

  An exception occurred while executing a query: SQLSTATE[42703]: Undefined column: 7 ERROR:  column "type" of relation "appconfig" does not exist
  LINE 1: UPDATE "appconfig" SET "type" = $1 WHERE ("appid" = $2) AND ...
                                 ^


In ExceptionConverter.php line 68:

  An exception occurred while executing a query: SQLSTATE[42703]: Undefined column: 7 ERROR:  column "type" of relation "appconfig" does not exist
  LINE 1: UPDATE "appconfig" SET "type" = $1 WHERE ("appid" = $2) AND ...
                                 ^


In Exception.php line 28:

  SQLSTATE[42703]: Undefined column: 7 ERROR:  column "type" of relation "appconfig" does not exist
  LINE 1: UPDATE "appconfig" SET "type" = $1 WHERE ("appid" = $2) AND ...
                                 ^


In Statement.php line 130:

  SQLSTATE[42703]: Undefined column: 7 ERROR:  column "type" of relation "appconfig" does not exist
  LINE 1: UPDATE "appconfig" SET "type" = $1 WHERE ("appid" = $2) AND ...
                                 ^


config:app:set [--output [OUTPUT]] [--value VALUE] [--type TYPE] [--lazy|--no-lazy] [--sensitive|--no-sensitive] [--update-only] [--] <app> <name>

root@caradhras:[/docker/nextcloud]:
```

Thanks a lot!

**Comment by kesselb** [comment]
> Is it related please?

Unrelated.

**Comment by aronmgv** [comment]
> > Is it related please?
> 
> Unrelated.

Should I create new issue then?

**Comment by kesselb** [comment]
> Should I create new issue then?

Yes please

**Comment by ariep** [comment]
We're blocked by this from upgrading our fleet from 28 to 29: we have a provisioning script which does a `php occ config:import ...`, and the json file contains among many other settings
```
[...]
        "apps":{                                                                                                                                                                              
            "core":{                                                                                                                                                                          
                "backgroundjobs_mode": "webcron"
[...]
```
so this bug makes the provisioning fail.

Will this bug be fixed in the 29 series? I see that the MR has milestone set to 31, but as you can't skip major versions that would be unfortunate.

**Comment by kesselb** [comment]
@ariep 

I don't think the pull request will address the problem with the config import command. 
Please leave a comment on the pull request, maybe that can be considered. 
Don't forget to also open a support ticket with Nextcloud if you have a support contract.

**Comment by mkleger** [comment]
I have the same problem when setting the background image via OCC and also via the WebUI with Nextcloud 30.0.0
![image](https://github.com/user-attachments/assets/cec12c65-996c-4e0d-b2cb-bcf69887b608)

**Comment by kesselb** [comment]
I'm pleasantly surprised to see the 18 people used the thumbs up reaction, thank you :+1: 

It's definitely annoying that changing the background image does is broken, sorry for that :disappointed: 

Could some please log a fresh issue for the problem? It's a different component that needs a different fix than the background mode. It's perfectly fine to omit most of the issue template (e.g. we don't need the configuration report, logs, etc). Just the basic steps to reproduce, your Nextcloud version and a screenshot would be nice.

Thanks everyone :+1:

**Comment by MajorTomDE** [comment]
Same issue here

**Comment by AnonymousX86** [comment]
Issue appeared after upgrade to `30.0.0`.
